### DISCUSSION TITLE
⭐ New Release: 3.3.0 Final

### DISCUSSION THREAD

**Discussion by Xanashi** [Xanashi/Icaros#39] [Releases]
Good Afternoon Everyone,

The final version of Icaros 3.3.0 is finally here.
I got through most of the showstoppers. You can find the few changes between this and the previous RC1 build in a post below.

A new build will be coming soon with some property handler improvements, so stay tuned!

Thank you so much to everyone you has provided feedback and helped me troubleshoot various issues!
I really appreciate the engagement and support! <3

Also thanks to Witchilich, you can now also find and install Icaros with [Winget](https://learn.microsoft.com/en-us/windows/package-manager/winget/)!

EDIT: For full transparency as always, here are the donation amount from 2022:

**2022 - $358**

Huge thanks to all who donated to keep the Icaros development alive and well! ❤️ 

I hope you all enjoy the new releases! 
<br/>
[![Icaros releases](https://img.shields.io/github/downloads/xanashi/icaros/v3.3.0/total?color=%232B6870&label=DOWNLOADS&style=for-the-badge)](https://github.com/Xanashi/Icaros/releases/tag/v3.3.0)

_______________________________________________________________________________________________________

### Icaros v.3.3.0 Final

**[Major Changes]**

- Added support for MP4 properties (incl. mov, 3gp, 3g2, mj2, psp, m4b, ism, ismv, f4v, m4a isma)
- Added support for AVIF thumbnailing
- Added support for TGA thumbnailing
- Added thumbnailing of AI files with an embedded image
- Added thumbnailing of PDF files with an embedded image
- Added AIF/AIFF/AIFC property support
- Added OGA, WEBMV and WEBMA properties
- Upgraded all Icaros Cache logic to 64bit
- Heavily improved all Icaros Cache logic 
- The Icaros Cache no longer has a size limit where it stops working
- Added several new thumbnail cache sizes
- Improved automatic thumbnail size detection in the Icaros Cache for certain filetypes
- Improved cover selection logic of CBZ/CBR/CB7/EPUB files
- Implemented a way for Icaros to recover slightly corrupted cache files

**[Fixes]**

- Automatically undo registry block placed by other software, that prevented Icaros from working
- Disable the 'Icons Only' File Explorer folder setting when enabling thumbnailing
- Fixed thumbnailing of Dolby Vision HEVC streams (dvhe)
- Fixed and improved the WAV property and tag parser
- Fixed rare Explorer hang on specific files
- Fixed thumbnailing of certain PAL8 files
- Fixed a couple of rare issues that affected thumbnailing of some files
- Fixed cache crash when caching very small image thumbnails
- Fixed multiple potential cache crashes
- Fixed caching of folder thumbnails on Windows 10+
- Fixed cache indexer not using the thumbnail timestamp when enabled
- Fixed issue where cache didn't always return a valid thumbnail even though one was available
- Fixed a lot of small cache issues
- Fixed crash if the current Icaros Cache Location had suddenly gotten
  restrictive permissions applied to the folder
- Fixed Cache Indexer returning an 0x14 error when parsing long filenames (> 259 characters)
- Fixed IcarosConfig not properly handling long filenames (> 259 characters)
- Fixed Boolean file properties showing incorrect values in IcarosConfig (Debug Page)
- Fixed crash when adding files without an extension to IcarosConfig

**[Improvements]**

- Improved Debug Mode: Icaros should now catch far fewer crashes caused
  by other software loaded in the same process as Icaros
- File Explorer settings that have been modified by Icaros is now reverted during uninstall
- Made the [Und] track language tag optional, and hidden by default
- IcarosConfig now performs a permission check on the current Icaros 
  Cache Location, when going to the Cache Page. If the check fails,
  an option has been added to attempt to fix the permission issues.
- Improved how IcarosConfig handles permission issues for the Icaros Cache
- When opening IcarosConfig for the first time, the default theme will now 
  match the current Windows App theme
- Added a workaround for an issue that prevented the Icaros Thumbnail Provider from working on Vista
- IcarosConfig now uses %SystemRoot% instead of the hardcoded C:\Windows path when referencing the WOW64 regsvr32.exe
- Added additional error reporting during cache indexing and for certain IcarosConfig actions
- All MP4 properties are disabled by default (As the native Windows handler allows editing of properties)
- MP3 properties are now disabled by default 
- Register OGA as an audio filetype
- Sidepanel position is now remembered in IcarosConfig
- Additional minor fixes and improvements

**[Localizations & Other]**

- Installer is now localized (Thank you bovirus for the implementation!) 
- Installer has been updated with a slightly more modern look
- Added Romanian localization (Thank you Cătălin Popescu! ^__^)
- Added Polish installer localization (Thank you Xyzzy! ^__^)
- Updated Polish localization (Thank you Xyzzy! ^__^)
- Updated Italian localization (Thank you bovirus! ^__^)
- Updated localizations (Simplified Chinese)  (Thank you ZerOri! ^__^)
- Updated VS, donations amount and copyright
- Updated LibDav1d to 1.0.0
- Updated Unarr to 1.1.0 Beta
- Updated GCC 
- Updated FFmpeg

**NOTE**: All your existing Icaros Cache files will be automatically upgraded with this build and
will therefore no longer work with previous versions of Icaros. Make sure you take a backup of 
these files if you're testing a Beta release of Icaros.

**NOTE**: If you decide to downgrade from this release, please note that your property settings in Icaros
will be reset back to the default values, and will need to be set again.

<hr /><em>This discussion was created from the release <a href='https://github.com/Xanashi/Icaros/releases/tag/v3.3.0'>Icaros 3.3.0</a>.</em>

**Comment by Xanashi** [comment]
**Changes between RC1 and the Final build**

- Added support for TGA thumbnailing
- Improved handling of long filenames in the IcarosCache
- Fixed IcarosConfig not properly handling long filenames
- Fixed expansion of %SystemRoot% in IcarosConfig
- Fixed Boolean file properties showing incorrect values in IcarosConfig (Debug Page)
- Fixed crash when adding files without an extension to IcarosConfig
- Fixed filename not showing up in the minidump dialog
- Added additional error reporting during cache indexing and for certain IcarosConfig actions
- Added Polish installer localization (Thank you Xyzzy! ^__^)
- Updated Polish localization (Thank you Xyzzy! ^__^)
- Updated FFmpeg

**Comment by Klaus1189** [comment]
Thank you very much ❤️❤️❤️
